.github/actions/cdk-deploy/action.yml

```
@@ -0,0 +1,92 @@
+name: Deploy
+
+inputs:
+  env_aws_secret_name:
+    required: false
+    type: string
+    default: ''
+  dir:
+    required: false
+    type: string
+    default: "."
+  script_path:
+    required: true
+    type: string
+    default: ''
+  skip_deploy:
+    required: false
+    type: string
+    default: 'false'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install node and related deps
+      uses: actions/setup-node@v4
+      with:
+        node-version: 20
+
+    - uses: actions/cache@v3
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+
+    - name: Install AWS CDK
+      shell: bash
+      run: npm install -g aws-cdk@2
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
+      with:
+        version: "0.5.*" 
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+
+    - name: Install dependencies
+      shell: bash
+      working-directory: ${{ inputs.dir }}
+      run: |
+        uv sync --only-group deployment
+        uv run npm install
+        uv run pip install boto3 # for the generate_env_file.py script
```

This might just be an alias for `uv run pip install` but `uv pip install` is the recommended way to do this

```suggestion
        uv pip install boto3 # for the generate_env_file.py script
```